index.js

```diff
@@ -149,42 +208,60 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+
+let num = 0
+function takeATicketNumber(line){
+  num++
+  line.push(num)
+  return `Welcome. Your are ticket number ${num}`
+}
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
-// console.log("%c----------", "color: red") 
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(takeANumber(line))
+// => `Welcome. You are ticket number 3`
 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+function ticketNumberGeneratorFunc(){
+  num = 0
+  function takeATicketNumberFunc(array){
+    num++
+    array.push(num)
+    return `Wwelcome. You are ticket number ${num}`
+  }
+  return takeATicketNumberFunc
+}
```

Comment:
Great work on this! Just be careful about declaring variables without let or const (line 244): 

```js
function ticketNumberGeneratorFunc(){
  let num = 0
  function takeATicketNumberFunc(array){
    num++
    array.push(num)
    return `Wwelcome. You are ticket number ${num}`
  }
  return takeATicketNumberFunc
}
```